rpc/handlers.go

```diff
@@ -409,3 +412,41 @@ func (h *Handler) StateUpdate(id *BlockID) (*StateUpdate, *jsonrpc.Error) {
 		},
 	}, nil
 }
+
+// Syncing returns the syncing status of the node.
+//
+// It follows the specification defined here:
+// https://github.com/starkware-libs/starknet-specs/blob/a789ccc3432c57777beceaa53a34a7ae2f25fda0/api/starknet_api_openrpc.json#L569
+func (h *Handler) Syncing() (*SyncState, *jsonrpc.Error) {
+	defaultSyncState := &SyncState{False: new(bool)}
+
+	startingBlockNumber := h.synchronizer.StartingBlockNumber
+	if startingBlockNumber == nil {
+		return defaultSyncState, nil
+	}
+	startingBlockHeader, err := h.bcReader.BlockHeaderByNumber(*startingBlockNumber)
+	if err != nil {
+		return defaultSyncState, nil
+	}
+	head, err := h.bcReader.HeadsHeader()
+	if err != nil {
+		return defaultSyncState, nil
+	}
```

Comment:
should these be returning an error instead of defaultSyncState?

Reply:
No. The spec doesn't specify an error. It should return false if it's not synching